src/main/java/uk/gov/hmcts/divorce/common/service/task/UpdateSuccessfulPaymentStatus.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.divorce.common.service.task;
+
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.type.ListValue;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.Payment;
+import uk.gov.hmcts.divorce.divorcecase.model.PaymentStatus;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+import static uk.gov.hmcts.ccd.sdk.type.YesOrNo.YES;
+
+@Component
+@Slf4j
+public class UpdateSuccessfulPaymentStatus implements CaseTask {
+
+    @Override
+    public CaseDetails<CaseData, State> apply(final CaseDetails<CaseData, State> caseDetails) {
+
+        log.info("Setting payment status to Success for Case ID: {}", caseDetails.getId());
+
+        List<ListValue<Payment>> applicationPayment =
+                caseDetails.getData().getApplication().getApplicationPayments();
```

Comment:
I think this will need to be different for respondent final orders, because the payment collection for those is FinalOrder.getFinalOrderPayments() rather than applicationPayments.

We could turn this into a service and pass in the payments object from outside, then it will work for either payments field and be reusable in the future if we add another payment type

Reply:
We discussed this on Slack and agreed to do it as case task for now